### PR TITLE
SpellChecker Interface

### PR DESCRIPTION
**Pull request by share-with-me**
The current logic of mapping the suggestions on the frontend is based up on indices. I have a better idea in mind where I would make use of `token` returned from backend to do a string comparison on the frontend. However, this PR till then, could be reviewed and improved upon. 

screenshot from 2017-07-27 00-06-24

screenshot from 2017-07-27 00-06-47

### CODE REVIEW

**Comment by share-with-me** [comment]
Travis is going to be very very unhappy. Please don't look at the build results :P

**Comment by share-with-me** [comment]
@sushain97 , Um, the Travis shows 2 errors. I am using `currentSpellCheckerRequest` in `callApy` but I do not understand the error. There is one more error about complexity 37. Could you help me in understanding what these are and how they can be resolved?

**Comment by sushain97** [comment]
I will be AFK till late this evening. You should be able to Google the
complexity error. It's an ESLint rule.

File: assets/css/style.css [hunk 1]
```diff
@@ -289,6 +289,36 @@ html[dir='rtl'] #srcLanguages {
...
+/* SpellCheker */
+
+#spellcheckerInput {
+    display: block;
+    height: 131px;
+    overflow: auto;
+    resize: both;
+}
+
+.spellcheckVisible .spellError {
```
Comment (sushain97) [inline assets/css/style.css]: `.spellcheckVisible` => `.spellcheckerVisible`

File: assets/css/style.css [hunk 2]
```diff
@@ -289,6 +289,36 @@ html[dir='rtl'] #srcLanguages {
...
+    border-bottom: 1px solid #f00;
+}
+
+.popover-content { /* sass-lint:disable-line class-name-format */
```
Comment (sushain97) [inline assets/css/style.css]: Can we make this selector more specific?

File: assets/css/style.css [hunk 3]
```diff
@@ -289,6 +289,36 @@ html[dir='rtl'] #srcLanguages {
     margin-left: 30px;
 }
 
+/* SpellCheker */
```
Comment (sushain97) [inline assets/css/style.css]: Two words also spelling :)

File: assets/css/style.css [hunk 4]
```diff
@@ -289,6 +289,36 @@ html[dir='rtl'] #srcLanguages {
...
+}
+
+.popover-content { /* sass-lint:disable-line class-name-format */
+    max-height: 190px;
+    overflow: auto;
+    padding: 0;
+    width: 110%;
+}
+
+.list-group { /* sass-lint:disable-line class-name-format */
```
Comment (sushain97) [inline assets/css/style.css]: Same with this and the next two. Basically my ask is to wrap it in a more specific class.

File: assets/js/spellchecker.js [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+
+if(modeEnabled('spellchecker')) {
+    $(document).ready(function () {
+        restoreChoices('spellchecker');
+        var timer, timeout = 2000;
+        $('#spellChekerForm').submit(function () {
+            clearTimeout(timer);
+            check();
+        });
+
```
Comment (sushain97) [inline assets/js/spellchecker.js]: extra line?

File: assets/js/spellchecker.js [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+    $('#spellcheckerInput').html('');
+    currentSpellCheckerRequest = callApy({
+        data: {
+            'q': words,
+            'lang': $('#primarySpellcheckerMode').val()
+        },
+        success: function (data) {
+            var originalWordsIndex = 0;
+            for(var tokenIndex = 0; tokenIndex < data.length; tokenIndex++) {
+                if(data[tokenIndex]['known'] === true) { // eslint-disable-line dot-notation
```
Comment (sushain97) [inline assets/js/spellchecker.js]: Why did you disable eslint here?

Comment (sushain97) [inline assets/js/spellchecker.js]: (and below)

File: assets/js/spellchecker.js [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+                    splitWords[originalWordsIndex] + '>' + splitWords[originalWordsIndex] + '</span>');
+                content[splitWords[originalWordsIndex]] = '<div class="list-group">';
+                for(var sugg = 0; sugg < data[tokenIndex]['sugg'].length; sugg++) { // eslint-disable-line dot-notation
+                    content[splitWords[originalWordsIndex]] += '<a href="#" class="list-group-item">' +
+                        data[tokenIndex]['sugg'][sugg][0] + '</a>'; // eslint-disable-line dot-notation
+                    content[splitWords[originalWordsIndex]] += '</div>';
+                }
+                $('.spellError').each(function () {
+                    var currentTokenId = this.id;
+                    $(this).popover({animation: false, placement: 'bottom', trigger: 'manual', html: true,
```
Comment (sushain97) [inline assets/js/spellchecker.js]: Each property per line

File: assets/js/spellchecker.js [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+    }
+
+    return deferred.promise();
+}
+
+if(modeEnabled('spellchecker')) {
+    $(document).ready(function () {
+        restoreChoices('spellchecker');
+        var timer, timeout = 2000;
+        $('#spellChekerForm').submit(function () {
```
Comment (JPJPJPOPOP) [inline assets/js/spellchecker.js]: `$('#spellChekerForm')` => `$('#spellCheckerForm')`

File: assets/css/style.css (hunk 2 above)
Comment (JPJPJPOPOP) [reply assets/css/style.css]: `.popover-content` doesn't seem to be used anywhere else in the code.

**Pull request closed without merging**